js/app.js

```diff
@@ -61,23 +70,34 @@ do {
             continue;
         }
 
-        if(action === '+') {
-            calc.add(number1, number2);
-        } else if(action === '-') {
-            calc.subtract(number1, number2);
-        } else if(action === '*') {
-            calc.multiply(number1, number2);
-        } else if(action === '/') {
-            calc.divide(number1, number2);
-        } else if(action === '^') {
-            calc.pow(number1, number2);
-        }  
-            
+        // if(action === '+') {
+        //     calc.add(number1, number2);
+        // } else if(action === '-') {
+        //     calc.subtract(number1, number2);
+        // } else if(action === '*') {
+        //     calc.multiply(number1, number2);
+        // } else if(action === '/') {
+        //     calc.divide(number1, number2);
+        // } else if(action === '^') {
+        //     calc.pow(number1, number2);
+        // }  
+
+       
+
+        // const operation = operations[action];
+        // if (operation) {
+        //     operation(num1, num2);
+        // }
+   
+
+        const result = operations[action](num1, num2);
```

Comment:
Warto jeszcze sprawdzić czy taka operacja istnieje tj.
```
const operation = operations[action];
if (operation) {
    operation(num1, num2);
}
```

PS. Jeśli nigdzie nie wykorzystujesz wyniku z funkcji to nie trzeba pisać `const result = `